### PR TITLE
gRPC service support + use SDK as Conan package

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
# Description

* Add support for using gRPC-based services as defined in app manifest
* Use C++ SDK as (local) Conan package (removing SDK subrepo)

## Issue ticket number and link

AB#16326 (task 16813)

## Checklist

* [ ] Vehicle App can be started with dapr run and is connecting to vehicle data broker
* [ ] Vehicle App can process MQTT messages and call the seat service
* [ ] Vehicle App can be deployed to local K3D and is running
* [ ] Created/updated tests, if necessary. Code Coverage percentage on new code shall be >= 70%.
* [ ] Extended the documentation in Velocitas repo
* [ ] Extended the documentation in README.md

* [ ] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [ ] Devcontainer can be re-built successfully

* [ ] Release workflow is passing

### CODE REVIEW

File: .velocitas.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
         },
         {
             "name": "devenv-devcontainer-setup",
-            "version": "v1.2.1"
+            "version": "grpc-interface-support"
```
Comment (BjoernAtBosch) [inline .velocitas.json]: Replace by final package version!

File: conanfile.txt [hunk 2]
```diff
@@ -1,4 +1,6 @@
 [requires]
+seats-service-sdk/auto
+vehicle-app-sdk/prep-rel-0.4.0
```
Comment (BjoernAtBosch) [inline conanfile.txt]: Replace once SDK is released:
```suggestion
vehicle-app-sdk/0.4.0
```

File: conanfile.txt [hunk 3]
```diff
@@ -1,4 +1,6 @@
 [requires]
+seats-service-sdk/auto
```
Comment (BjoernAtBosch) [inline conanfile.txt]: @doosuu: Should `auto` be replaced?

File: app/AppManifest.json [hunk 4]
```diff
@@ -27,6 +27,13 @@
                     "sampleapp/currentSpeed"
                 ]
             }
+        },
+        {
+            "type": "grpc-interface",
+            "config": {
+                "src": "https://raw.githubusercontent.com/eclipse/kuksa.val.services/main/seat_service/proto/sdv/edge/comfort/seats/v1/seats.proto",
+                "requires": { }
+            }
```
Comment (BjoernAtBosch) [inline app/AppManifest.json]: Do we want to keep this? Actually, it's not used yet.

Comment (doosuu) [reply app/AppManifest.json]: SHould be removed

File: conanfile.txt (hunk 3 above)
Comment (doosuu) [reply conanfile.txt]: Any suggestions? `generated` might also be an option

**Review by BjoernAtBosch — COMMENTED** [review]
LGTM (but I cannot approve because I did contribute)

**Review by doosuu — DISMISSED** [review]
LGTM

File: app/src/CMakeLists.txt [hunk 5]
```diff
@@ -19,10 +19,6 @@ add_executable(${TARGET_NAME}
     Launcher.cpp
 )
 
-add_dependencies(${TARGET_NAME}
-    vehicle-app-sdk
-)
-
 target_link_libraries(${TARGET_NAME}
-    vehicle-app-sdk
+    ${CONAN_LIBS}
```
Comment (MP91) [inline app/src/CMakeLists.txt:23]: is this just the sdk or are we now linking against all conan-libs which we install?

Comment (MP91) [reply app/src/CMakeLists.txt:23]: all unused dependencies are stripped when building the container

**Review by MP91 — DISMISSED** [review]
LGTM

**Review by MP91 — APPROVED** [review]
SLGTM

**Pull request merged**